models/AbstractTreeGenerator.php

```diff
@@ -0,0 +1,234 @@
+<?php
```

Comment:
@boajer Do you think an `action`, `controller` or a `command` should have all this logic inside or just forward parameters provided by `input` to `specialised services` that can be reusable and splitted? 

Same applies to `MediaTreeGenerator` https://github.com/oat-sa/extension-tao-devtools/pull/187/files#diff-145a1cf8a6283b9ee08e905ced2ed6feabc9eae50d30eaa12963b7b068edf894R30

Reply:
Yep, I agree; This time i've just modified a bit old script to simplify asset generation; I would go with bigger refactoring next time

Reply:
I would consider extracting this logic to separate service as a scope of this PR, same applies to MediaTreeGenerator.

Reply:
IMO since you modified abstraction and created new implementations I think the work is in the scope of this task. Maybe when we decide to extract / refactor we are having different perceptions within the team. 

Although I cannot say it is a mandatory requirement, I strongly recommend doing this to improve / facilitate testing, code reusability and maintenance. If you feel it does not fit here, just close my comment :) 

I understand the time limitations :)